app/kafka/manager/actor/cluster/KafkaStateActor.scala

```diff
@@ -1299,5 +1315,104 @@ class KafkaStateActor(config: KafkaStateActorConfig) extends BaseClusterQueryCom
     Option(fn(topicsTreeCache))
   }
 
+  //---------------------------------------------------
+  private[this] var kafkaTopicOffsetGetter : Option[KafkaTopicOffsetGetter] = None 
+  private[this] var kafkaTopicOffsetMap = new TrieMap[String, Map[Int, Long]]
+  private[this] var kafkaTopicOffsetCaptureMap = new TrieMap[String, PartitionOffsetsCapture]
+  def startTopicOffsetGetter() : Unit = {
+    log.info("Starting kafka managed Topic Offset Getter ...")
+    kafkaTopicOffsetGetter = Option(new KafkaTopicOffsetGetter())
+    val topicOffsetGetterThread = new Thread(kafkaTopicOffsetGetter.get, "KafkaTopicOffsetGetter")
+    topicOffsetGetterThread.start()
+  }
+
+  def stopTopicOffsetGetter() : Unit = {
+    kafkaTopicOffsetGetter.foreach {
+      kto =>
+        Try {
+          log.info("Stopping kafka managed Topic Offset Getter ...")
+          kto.close()
+        }
+    }
+  }
+
+  class KafkaTopicOffsetGetter() extends Runnable {
+    @volatile
+    private[this] var shutdown: Boolean = false
+
+    override def run(): Unit = {
+      while (!shutdown) {
+        try {
+          withTopicsTreeCache {  cache: TreeCache =>
+            cache.getCurrentChildren(ZkUtils.BrokerTopicsPath)
+            }.fold {
+              } { data: java.util.Map[String, ChildData] =>
+                var broker2TopicPartitionMap: Map[BrokerIdentity, List[(TopicAndPartition, PartitionOffsetRequestInfo)]] = Map()
+
+                data.asScala.keys.toIndexedSeq.foreach(topic => {
+                  var optPartitionsWithLeaders : Option[List[(Int, Option[BrokerIdentity])]] = getPartitionLeaders(topic)
```

Comment:
Maybe add another check here for shutdown.  If the list of topics is long, it may take some time to shutdown.